src/main/java/seedu/address/model/person/UniqueLogList.java

```diff
@@ -0,0 +1,132 @@
+
+    /**
+     * Returns true if the list contains an equivalent log as the given argument.
+     */
+    public boolean contains(Log log) {
+        requireAllNonNull(log);
+        return this.internalList.stream().anyMatch(log::isSameLog);
+    }
+
+    /**
+     * Returns true if the list contains a log that matches exactly the
+     * given {@code Log}. A stronger form of contains.
+     */
+    public boolean containsExactly(Log log) {
+        requireAllNonNull(log);
+        return this.internalList.stream().anyMatch(log::equals);
+    }
+
+    /**
+     * Adds a log to the list.
+     * An identical log must not already exist in the list.
+     */
+    public void addLog(Log log) {
+        requireNonNull(log);
+        if (this.contains(log)) {
+            throw new DuplicateLogException();
+        }
+        internalList.add(log);
+    }
+
+    /**
+     * Replaces the {@code target} in the list with the {@code editedLog}.
+     * The {@code target} log must exist in the list, and the identity of the {@code editedLog}
+     * must be different from all existing logs in the list.
+     */
+    public void setLog(Log targetLog, Log editedLog) {
+        requireAllNonNull(targetLog, editedLog);
+
+        if (!targetLog.isSameLog(editedLog) && this.contains(editedLog)) { // trying to insert something already inside
+            throw new DuplicateLogException();
+        }
+
+        int index = this.internalList.indexOf(targetLog);
+        if (index == -1) {
+            throw new LogNotFoundException();
+        }
+
+        this.internalList.set(index, editedLog);
+    }
+
+    /**
+     * Removes the equivalent log from the list.
+     * The log must exist in the list.
+     */
+    public void removeLog(Log toRemove) {
+        requireNonNull(toRemove);
+        if (!this.internalList.remove(toRemove)) {
+            throw new LogNotFoundException();
+        }
+    }
+
+    /**
+     * Replaces the contents of the list with all of {@code logs}.
+     * {@code logs} must not contain duplicate log.
+     */
+    public void setLogs(List<Log> logs) {
+        requireAllNonNull(logs);
+        if (!this.logsAreUnique(logs)) {
+            throw new DuplicateLogException();
+        }
+        this.internalList.setAll(logs);
+    }
+
+    /**
+     * Returns true if and only if {@code logs} contains only unique (by title) logs.
+     */
+    public boolean logsAreUnique(List<Log> logs) {
+        int n = logs.size();
+        for (int i = 0; i < n - 1; i++) { // iterate like N choose 2
+            for (int j = i + 1; j < n; j++) {
+                if (logs.get(i).isSameLog(logs.get(j))) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Returns the list of logs as an unmodifiable {@code ObservableList}.
+     */
+    public ObservableList<Log> asUnmodifiableObservableList() {
+        return this.internalUnmodifiableList;
+    }
```

Comment:
I think this method should return an updated internalUnmodifiableList as if the internalist is updated, the internUnmodifiableList will not be updated as well, thus any changes will not be reflected in the UI.

So maybe something like `return FXCollections.unmodifiableObservableList(internalList);`

Reply:
Why?

Note that the object being returned is already `FXCollections.unmodifiableObservableList(internalList)`.

Reply:
Yes you are right! I realised FXCollections.unmodifiableObservableList(internalList), uses ponter to the internalList.